### DISCUSSION TITLE
Text gets selected when pasted into a terminal emulator

### DISCUSSION THREAD

**Discussion by tsujan** [lxqt/lxqt#1897] [Q&A] [answered]
That's started recently, after an upgrade, and it happens in QTerminal, xterm and konsole (I think it happens in all terminal emulators).

It's good that the pasted text is selected because it's distinguished from other texts. But I have a few questions:

Do you know which update causes it (bash, readline,...)? Is it configurable? Does it have a special name?

**Comment by tsujan** [comment]
Found a partial answer: It's called "bracketed paste mode" and is implemented in `TerminalDisplay::bracketText`.

Why it didn't work before, I don't know; it just started to work here after an upgrade, while I hadn't changed anything.

I'll add an experimental setting for it and make a PR if it works fine.

**Comment by Chiitoo** [comment, marked as answer]
I believe it's a change in readline 8.1 (default changed).

1. https://tiswww.case.edu/php/chet/readline/CHANGES

**Reply by tsujan** [reply]
Yes, it seems so. `readline` was among the upgraded packages here.

Anyway, I'll add an experimental option: "Disable bracketed paste mode"

EDIT: I added it and it worked fine:

![bracketed_paste](https://user-images.githubusercontent.com/981076/102718976-7233f200-4300-11eb-9a6f-f7edd1b84fee.png)

Will make 2 PRs soon.

**Reply by tsujan** [reply]
The PRs: https://github.com/lxqt/qtermwidget/pull/390 → https://github.com/lxqt/qterminal/pull/772
